test/system/use_worker_pool_tests.rb

```diff
@@ -46,108 +50,107 @@ class UseWorkerPoolTests < SystemTests
         :num_workers  => 2,
-        :logger       => TEST_LOGGER,
-        :do_work_proc => proc{ |work| sleep(work) }
+        :logger       => TEST_LOGGER
       })
-      @work_pool.start
+      @worker_pool.start
     end
 
     should "spawn and wait until work is available" do
       # the workers should be spawned and waiting
-      assert_equal 2,    @work_pool.waiting
-      assert_equal true, @work_pool.worker_available?
+      assert_equal 2,    subject.waiting
+      assert_equal true, subject.worker_available?
 
       # only one worker should be waiting
-      @work_pool.add_work 5
-      assert_equal 1,    @work_pool.waiting
-      assert_equal true, @work_pool.worker_available?
+      subject.add_work 5
+      assert_equal 1,    subject.waiting
+      assert_equal true, subject.worker_available?
 
       # neither worker should be waiting now
-      @work_pool.add_work 5
-      assert_equal 0,     @work_pool.waiting
-      assert_equal false, @work_pool.worker_available?
+      subject.add_work 5
+      assert_equal 0,     subject.waiting
+      assert_equal false, subject.worker_available?
     end
 
     should "go back to waiting when they finish working" do
-      assert_equal 2, @work_pool.waiting
-      @work_pool.add_work 1
-      assert_equal 1, @work_pool.waiting
+      assert_equal 2, subject.waiting
+      subject.add_work 1
+      assert_equal 1, subject.waiting
 
       sleep 1 # allow the worker to run
 
-      assert_equal 2, @work_pool.waiting
+      assert_equal 2, subject.waiting
     end
 
   end
 
   class WorkerCallbackTests < SystemTests
     setup do
-      @error_called       = false
-      @start_called       = false
-      @shutdown_called    = false
-      @sleep_called       = false
-      @wakeup_called      = false
-      @before_work_called = false
-      @after_work_called  = false
-
-      @worker_pool = DatWorkerPool.new({
-        :do_work_proc => proc{ |work| raise work if work == 'error' }
-      })
+      @callbacks_called = {}
 
-      @worker_pool.on_worker_error{ @error_called = true }
-      @worker_pool.on_worker_start{ @start_called = true }
-      @worker_pool.on_worker_shutdown{ @shutdown_called = true }
-      @worker_pool.on_worker_sleep{ @sleep_called = true }
-      @worker_pool.on_worker_wakeup{ @wakeup_called = true }
-      @worker_pool.before_work{ @before_work_called = true }
-      @worker_pool.after_work{ @after_work_called = true }
-    end
-    teardown do
-      # TODO - remove once a worker class can be passed
-      DefaultWorker.on_start_callbacks.clear
-      DefaultWorker.on_shutdown_callbacks.clear
-      DefaultWorker.on_sleep_callbacks.clear
-      DefaultWorker.on_wakeup_callbacks.clear
-      DefaultWorker.on_error_callbacks.clear
-      DefaultWorker.before_work_callbacks.clear
-      DefaultWorker.after_work_callbacks.clear
+      @worker_class = Class.new do
+        include DatWorkerPool::Worker
+
+        # change once these are instance evald
```

Comment:
@jcredding did you mean to add "TODO" here?  yeah, it would be nice to use instance vars here instead of monkeying  with the params.